apps/go-mod-validator/test/github.test.ts

```diff
@@ -0,0 +1,114 @@
+import { isGoModReferencingDefaultBranch } from "../src/github";
+import { describe, expect, it, vi } from "vitest";
+import listTagsFixture from "./data/list_tags.json";
+import { GoModule } from "../src/deps";
+
+const owner = "smartcontractkit";
+const repo = "go-plugin";
+const goModPath = `github.com/${owner}/${repo}`;
+const mockGet = vi.fn();
+const mockCompareCommits = vi.fn();
+const mockListTags = vi.fn();
+const mockOctokit: any = {
+  rest: {
+    repos: {
+      get: mockGet,
+      compareCommits: mockCompareCommits,
+      listTags: mockListTags,
+    },
+  },
+};
+describe("isGoModReferencingDefaultBranch", () => {
+  describe("pseudo-versions", () => {
```

Comment:
Discussed offline, we could do this with actual commits and fixtured responses. Can be done in separate PR though.

Reply:
https://smartcontract-it.atlassian.net/jira/software/c/projects/RE/boards/83?selectedIssue=RE-2792